packages/content-conformance/src/rules/ensure-valid-nav-paths.js

```diff
@@ -0,0 +1,78 @@
+import flat from 'flat'
+
+// flatten navData json object and visit all paths (keys that end with ".path")
+//
+// ensure a <path>.mdx or <path>/index.mdx exists, but not both
+function visitor(navData, filePaths, report) {
+  // flatten nav data file
+  const flatNodes = flat(navData)
+  // get all the paths from the nav data file
+  const navDataPaths = Object.entries(flatNodes)
+    .filter(([key]) => {
+      return key.endsWith('.path')
+    })
+    .map(([, value]) => value)
+
+  navDataPaths.forEach((p) => {
+    const namedFile = `${p}.mdx`
+    const indexFile = p ? `${p}/index.mdx` : `index.mdx`
+
+    const namedFileExists = filePaths.includes(namedFile)
+    const indexFileExists = filePaths.includes(indexFile)
+
+    if (!namedFileExists && !indexFileExists) {
+      report(
+        `NavData path (${p}) is missing a corresponding file in the content directory`
```

Comment:
Is there a way we can include the expected file path in the message? I think that would make it more actionable for contributors.

Reply:
Not sold on the implementation, but these error messages should be significantly more actionable
- https://github.com/hashicorp/web-platform-packages/pull/137/commits/3fbd085ec12c93491f1d0a80c82dbc1e02ea3d39